### DISCUSSION TITLE
Improve file transfer speed

### DISCUSSION THREAD

**Discussion by liqijian101** [luc-github/ESP3D-TFT#71] [General]
I found a way to improve file upload speed，
https://github.com/espressif/esp-idf/issues/10493#issuecomment-1420561542

> If you're using this SPI peripheral only with one device (sd slot + card), then you can try to set CS in slot_config.gpio_cs to SDSPI_SLOT_NO_CS / GPIO_NUM_NC / -1.

**Comment by luc-github** [comment]
interresting , will you a try ?

**Reply by liqijian101** [reply]
I have already tested it。For me, the upload speed has increased from 180Kb/s to 400Kb/s

**Reply by luc-github** [reply]
wow impressive

**Reply by liqijian101** [reply]
I tried another identical ESP3D board, but it didn't work. I will continue to investigate the reason

**Comment by liqijian101** [comment]
I have found that the upload speed is closely related to the channel of the router.

**Reply by luc-github** [reply]
which channel is good ? and which channel  is bad ?

**Comment by liqijian101** [comment]
test result

![QQ图片20240503204140](https://github.com/luc-github/ESP3D-TFT/assets/16173343/353b4b63-986d-4e59-8bb6-eb76ae181a6a)

channel  13 ，400Kb/s
channel  10，very low

**Reply by luc-github** [reply]
can you compare with channel 1 ?
it could be an overlap with others channels (9/11) that disturb channel 10

**Reply by liqijian101** [reply]
about 200KB/s。

**Comment by liqijian101** [comment]
I will test channel 1 when I have the opportunity.
Currently, I have found that the main influencing factor is the router channel. Even without modifying the CS pins, a speed of 450K/B can still be achieved

**Reply by luc-github** [reply]
Here what I got as feedback from my IA assistant:

Channels 1, 6 and 11 are considered the best choices in the 2.4 GHz band.

In the 2.4 GHz band, there are 11 channels numbered 1 to 11 (in most countries). However, these channels are not completely separate from each other. Each channel has a bandwidth of 20 MHz, but they are only 5 MHz apart. This means that adjacent channels overlap and can interfere with each other.

Here's how the channels overlap:

Channel 1 overlaps channels 2, 3, 4 and 5
Channel 6 overlaps channels 3, 4, 5, 7, 8 and 9
Channel 11 overlaps channels 7, 8, 9 and 10
However, channels 1, 6 and 11 are the only ones that do not overlap with each other. This is why they are recommended - they allow you to have three Wi-Fi networks in the same area without direct interference between them.

So, although channel 6 is adjacent to channels 5 and 7, and channel 11 is adjacent to channels 10 and 12 (where 12 is not used in North America), these overlaps are less problematic than if they directly overlapped channels 1 or 6.

In summary, channels 1, 6 and 11 are the most spaced apart in the 2.4 GHz band, minimizing interference between them. This is why they are generally recommended for getting the best possible Wi-Fi performance.

**Reply by liqijian101** [reply]
In China, we use channels 1-13.
We should test which channel is most suitable and choose the most suitable channel

**Comment by liqijian101** [comment]
In SDIO mode, the upload speed can reach 690KB/s

**Comment by liqijian101** [comment]
After testing, in SDIO mode, the maximum file upload speed is about 800KB/s
And there is not much difference between the 1-bit bus width and the 4-bit bus width.
I think the bandwidth of SD cards is relatively high. The current upload speed may be influenced by the network, rather than the write speed of SD

**Reply by luc-github** [reply]
what method are you using for upload ? http ?

**Reply by liqijian101** [reply]
I upload it directly on the WEBUI。
<img width="312" alt="无标题" src="https://github.com/luc-github/ESP3D-TFT/assets/16173343/23edd2c9-5d23-425d-9e35-4de391c6ac7e">

**Reply by luc-github** [reply]
Ok in that case using webdav may increase speed a little I think

**Reply by liqijian101** [reply]
However, it is not。Webdav is even slower,about 780KB/s

**Reply by luc-github** [reply]
really ? interresting - it means my code on decoding multipart upload is better than the arduino esp32 webserver version ^_^

About SDIO 4 bits should be faster than 1 bit that is weird - according https://esp32.com/viewtopic.php?t=12365 the 984KB/S should be achievable with some tuning  - I remember also RRF porting team claimed they reached 1.2MB/S

**Reply by liqijian101** [reply]
The current speed is already very good

**Reply by luc-github** [reply]
Yes your finding are great ^_^

**Reply by liqijian101** [reply]
After all, different testing environments will have different results
